### PR TITLE
Support for explicit sync

### PR DESCRIPTION
**Pull request by Joshua-Ashton**

### CODE REVIEW

**Comment by KyunLFA** [comment]
Hi, if you don't mind my intrusion, I've tested this on my machine and with --expose-wayland it fails to render and hangs with "Not top level surface."... at least that's on my end, does it work with native Wayland on your end or, although not marked as draft, was this still a WIP?

I have a "hack" to make proceed to render, by forcing the toplevel role shortly into the xdg_surface_new call (since [this commit](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/c85838892d56111809aa2edb83a2f22428bfa806) surfaces, toplevels and also popups have been changed, this is the first Wayland breakage). I haven't noticed any regressions from doing this, but I'm also not very confident it's the right approach.

After getting assigning the toplevel role to the xdg surface, we need to schedule the surface configure which is no longer automatic on creation since [this commit](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/811ca199c444525dc4d846d38f76554f1a9b48b0) (second Wayland backend break). 

(Sidenote)
I've been sitting on some assortment of local changes I'm only now slowly pushing to my fork (I'll be posting my wlroots "hackjob" shortly), let me know if any of them are of use at all (of note, the initial Wine Wayland driver support, which funnily enough was very straightforward), and specially if some of them can manage as "passable" draft PRs, I say this as I'm still very amateur and early into software development.

Thank you for your continued contributions, cheers.

**Comment by Joshua-Ashton** [comment]
Hello. The --expose-wayland stuff is very very experimental. It does not surprise me that it doesn't work very well. It really just exists to get Waydroid to work.

I'd love to do more for that, but have not had the time.

You can test this MR by launching any Vulkan app **without** `--expose-wayland`, as the Gamescope WSI layer will forward to Wayland WSI.

**Comment by Joshua-Ashton** [comment]
Ah I see you had a commit there to try and fix xdg-shell stuff after the bump. I will merge that in.

File: src/wlserver.hpp [hunk 1]
```diff
@@ -38,6 +38,21 @@ struct wlserver_vk_swapchain_feedback
...
+
+struct GamescopeTimelinePoint
+{
+	struct wlr_render_timeline *pTimeline = nullptr;
+	uint64_t ulPoint = 0;
+
+	void Release();
+};
+
+struct GamesopeAcquireTimelineState
```
Comment (niklaskorz) [inline src/wlserver.hpp:50]:
```suggestion
struct GamescopeAcquireTimelineState
```

typo :)

**Pull request merged**